index.js

```diff
@@ -129,40 +130,54 @@ NYC.prototype.addAllFiles = function () {
 NYC.prototype._wrapRequire = function () {
   var _this = this
 
-  var babelRequireHook = null
-  var requireHook = function (module, filename) {
-    // allow babel's compile hoook to compile
-    // the code -- ignore node_modules, this
-    // helps avoid cyclical require behavior.
-    var content = null
-    if (babelRequireHook && filename.indexOf('node_modules/') === -1) {
-      babelRequireHook({
-        _compile: function (compiledSrc) {
-          _this.sourceMapCache.add(filename, compiledSrc)
-          content = compiledSrc
-        }
-      }, filename)
-    }
+  var defaultHook = function (module, filename) {
+    // instrument the required file.
+    var obj = _this.addFile(filename, false)
 
-    // now instrument the compiled code.
-    var obj = null
-    if (content) {
-      obj = _this.addContent(filename, content)
-    } else {
-      obj = _this.addFile(filename, false)
-    }
+    // always use node's original _compile method to compile the instrumented
+    // code. if a custom hook invoked the default hook the code should not be
+    // compiled using the custom hook.
+    Module.prototype._compile.call(module, obj.content, filename)
+  }
 
-    module._compile(obj.content, filename)
+  var wrapCustomHook = function (hook) {
+    return function (module, filename) {
```

Comment:
I love this approach, much cleaner than what I was doing before -- also I think that it fixes the issue I was originally having that required me to ignore the `node_modules` folder -- great job.